src/components/PaletteForm/PaletteForm.js

```diff
@@ -8,32 +8,75 @@ class PaletteForm extends Component {
       colors: [],
       newProjectName: '',
       newPaletteName: '',
-      selectedProjectName: ''
+      selectedProjectId: null,
     }
   }
 
-  // componentDidMount() {
-  //   // method for getting 5 random colors to set state
-  // }
+  componentDidMount() {
+    let colors = [];
+    while (colors.length < 5) {
+      colors.push(this.getRandomColor())
+    }
+    this.setState({ colors })
+  }
+
+  getRandomColor() {
+    return "#000000".replace(/0/g,() => {return (~~(Math.random()*16)).toString(16);});
+  }
+
+  updateColors(e) {
+    let colors = [];
+    while (colors.length < 5) {
+      colors.push(this.getRandomColor())
+    }
+    this.setState({ colors })
+  }
+
+  handleDropDownChange(e) {
+    // add error handling for not selected
+
+    this.setState({ selectedProjectId: e.target.value});
+  }
+
+  handleInputChange(e) {
```

Comment:
For the binding issue, I believe if you change this to ES6 syntax, handleInputChange = (e) => {...}, you will not have to include .bind anywhere. I would recommend just changing all these methods to ES6 syntax.

Reply:
Gotcha, I thought I had tried that but must have had something funky with my syntax. I will do that when I get back to it shortly.